litho-widget/src/main/java/com/facebook/litho/widget/RecyclerBinder.java

```diff
@@ -1122,6 +1122,44 @@ private void addToCurrentBatch(AsyncOperation operation) {
     mCurrentBatch.mOperations.add(operation);
   }
 
+  /**
+   * Replaces all items in the {@link RecyclerBinder} with the provided {@link RenderInfo}s.
+   */
+  @UiThread
+  public final void replaceAll(List<RenderInfo> renderInfos) {
+    synchronized (this) {
+      if (mHasAsyncOperations) {
+        throw new RuntimeException(
+            "Trying to do a sync replaceAll when using asynchronous mutations!");
+      }
+      final int oldSize = mComponentTreeHolders.size();
+      final int newSize = renderInfos.size();
+
+      if (oldSize > newSize) {
+        mComponentTreeHolders.subList(newSize, oldSize).clear();
+      }
+
+      for (int i = 0; i < newSize; i++) {
+        RenderInfo newRenderInfo = renderInfos.get(i);
+        if (i >= mComponentTreeHolders.size()) {
+          mComponentTreeHolders.add(i, createComponentTreeHolder(newRenderInfo));
+        } else {
+          ComponentTreeHolder holder = mComponentTreeHolders.get(i);
+          if (holder.getRenderInfo().rendersView() && newRenderInfo.rendersView()) {
+            // We can reuse the old holder if we're still just rendering Views.
```

Comment:
Is this beneficial somehow?

Reply:
Less allocations, but probably not significant enough to matter. I'll just change it to a simpler full replace.